extensions/amp-facebook-comments/1.0/facebook-comments.js

```diff
@@ -0,0 +1,45 @@
+import * as Preact from '../../../src/preact';
+import {useResourcesNotify} from '../../../src/preact/utils';
+import {getIframeProps} from '../../../src/3p-frame';
+import { ContainWrapper } from "../../../src/preact/component";
+
+const TAG = "AMP-FACEBOOK-COMMENTS";
+
+
+// TODO: After adding to type, do FB comments props
+/**
+ * @param {!FacebookCommentsProps} props
+ * @return {PreactDef.Renderable}
+ */
+export function FacebookComments(props) {
+  useResourcesNotify();
+  const locale = props['data-locale'] ? props['data-locale'] : 'EN_US';
+  let passedAttributes = Object.assign({}, datasetFromProps(props));
+  passedAttributes['width'] = props['width'];
+  passedAttributes['height'] = props['height'];
+  const iframeProps = getIframeProps(
+    window,
+    document.body,
+    passedAttributes,
+    "facebook",
+    undefined,
+    TAG,
+  );
+
+  document.body.append(document.createElement('script').setAttribute('src', 'https://connect.facebook.net/' + locale + '/sdk.js'));
```

Comment:
Would this attempt to append a script each time the function is invoked?

Reply:
Yes it will. That'll be an issue. Two solutions I can think of: 

(1) iterating through the document.body's elements to check if this script exists, and if it does, don't append. 
(2) Giving it a generic ID (i.e. "amp-facebook-sdk") and simply checking if such an ID exists before adding. I think this one probably has better performance.

In either of these cases, would any security issues arise?

Reply:
I'd recommend looking into a `useScript` hook or similar.